### PR TITLE
Include sources only if they exist

### PR DESCRIPTION
**Pull request by viranch**
Addresses #1066

### CODE REVIEW

**Review by ross — COMMENTED** [review]
Not at a point where I can sit down to run this, but main question here is whether this would change the previous behavior, i.e. throwing an error when a configured zone file doesn't exist.

**Comment by viranch** [comment]
The previous behavior would eventually end up throwing `FileNotFoundError` at some point when a zone file/dir doesn't exist since that case is not explicitly handled.

**Comment by ross** [comment]
> The previous behavior would eventually end up throwing `FileNotFoundError` at some point when a zone file/dir doesn't exist since that case is not explicitly handled.

Was guessing that since this is checking for existance that would change and instead of the user seeing an error they'd end up with an empty zone.

**Comment by viranch** [comment]
True, post-PR we'd be logging and continuing with an empty zone as opposed to exiting. I guess exiting with an error is the "right" behavior: if the zone already exists it'd be pretty dangerous to try and empty it even though octodns-sync would show sufficient deletes in the plan output.

File: octodns/provider/yaml.py [hunk 1]
```diff
@@ -353,9 +353,7 @@ def populate(self, zone, target=False, lenient=False):
             sources.append(join(self.directory, self.shared_filename))
 
         if not sources:
-            self.log.info(
-                'populate:   no YAMLs found for %s', zone.decoded_name
-            )
+            raise ProviderException(f'no YAMLs found for {zone.decoded_name}')
```
Comment (ross) [inline octodns/provider/yaml.py:356]: :cool:, That was the suggestion I came up with when thinking about it.

**Review by ross — APPROVED** [review]

**Comment by viranch** [comment]
> Throw a changelog entry in with this if you don't mind. I assume after it (and any other YAML fixes) land we'll cut a point release.

Added changelog entry. Yep hoping to cut a release after the current batch of yaml fixes are in.

**Comment by ross** [comment]
> Added changelog entry. Yep hoping to cut a release after the current batch of yaml fixes are in.

@viranch noticed you added a specific version & date in the CHANGELOG update. Confused me when I was looking at release versions just now (and when i made a changelog entry earlier. I though you'd already cut 1.1.2. We should probably do that soon and I guess put my RR fixes in and makes it 1.2.0?

**Comment by viranch** [comment]
> > Added changelog entry. Yep hoping to cut a release after the current batch of yaml fixes are in.
> 
> @viranch noticed you added a specific version & date in the CHANGELOG update. Confused me when I was looking at release versions just now (and when i made a changelog entry earlier. I though you'd already cut 1.1.2. We should probably do that soon and I guess put my RR fixes in and makes it 1.2.0?

Yep sorry my bad. You want to cut 1.2.0 instead of 1.1.2 or make them both separate releases?

**Comment by viranch** [comment]
> > Added changelog entry. Yep hoping to cut a release after the current batch of yaml fixes are in.
> 
> @viranch noticed you added a specific version & date in the CHANGELOG update. Confused me when I was looking at release versions just now (and when i made a changelog entry earlier. I though you'd already cut 1.1.2. We should probably do that soon and I guess put my RR fixes in and makes it 1.2.0?

Yep sorry my bad. You want to cut 1.1.2 then 1.2.0 or straight to 1.2.0?

**Comment by ross** [comment]
> Yep sorry my bad. You want to cut 1.1.2 then 1.2.0 or straight to 1.2.0?

I'd say straight to 1.2.0. Seems fine to mix patch fixes with the RR improvements since they shouldn't hurt anything.

**Pull request merged**